### PR TITLE
Enable authentication with any address type

### PR DESCRIPTION
**Pull request by gdethier**
* Add account types encapsulating account validation and conveying address type
* Enable authentication with account of any type
* Not yet tested against new backend (not yet available) so will require further adjustments

https://github.com/logion-network/logion-internal/issues/852

### CODE REVIEW

File: packages/client/src/Signer.ts [hunk 1]
```diff
@@ -174,8 +174,12 @@ export class KeyringSigner extends BaseSigner {
 
     private keyring: Keyring;
 
-    async signToHex(signerId: string, message: string): Promise<TypedSignature> {
-        const keypair = this.keyring.getPair(signerId);
+    async signToHex(signerId: ValidAccountId, message: string): Promise<TypedSignature> {
+        if(signerId.type !== "Polkadot") {
+            throw new Error(`Unsupported account type ${signerId.type}`);
+        }
```
Comment (benoitdevos) [inline packages/client/src/Signer.ts:180]: Correct me if I am wrong but I think this will prevent user to sign with Metamask.

**Review by benoitdevos — APPROVED** [review]
Good job !
Just one remark:

File: packages/client/src/Signer.ts (hunk 1 above)
Comment (gdethier) [reply packages/client/src/Signer.ts:180]: Yes, because that part still needs to be written.

Comment (benoitdevos) [reply packages/client/src/Signer.ts:180]: But it is already used when authenticating an Ethereum address with MetaMask in the certificate/restricted delivery. As such, this commit introduces a regression.

Comment (gdethier) [reply packages/client/src/Signer.ts:180]: Indeed, I misread that. I remove that part.

**Pull request merged**